cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControl.java

```diff
@@ -671,6 +669,17 @@ private void sanityCheckHardGoalPresence(List<String> goals, boolean skipHardGoa
     }
   }
 
+  /**
+   * Sanity check whether the given goals exist in the given supported goals.
+   * @param goals A list of goals.
+   * @param supportedGoals Supported goals.
+   */
+  private void sanityCheckNonExistingGoal(List<String> goals, Map<String, Goal> supportedGoals) {
+    goals.stream().filter(goalName -> supportedGoals.get(goalName) == null).forEach(goalName -> {
+      throw new IllegalArgumentException("Goal: " + goalName + " does not exist. Supported: " + supportedGoals.keySet());
+    });
+  }
+
```

Comment:
Here does the exception throws when first non existent goal is found? I feel it would be better to return full list of invalid goals.